pandas/core/indexes/category.py

```diff
@@ -659,6 +661,14 @@ def map(self, mapper):
 
     def _concat(self, to_concat: List["Index"], name: Label) -> "CategoricalIndex":
         # if calling index is category, don't check dtype of others
+        count = 0
+        for c in to_concat:
+            if not to_concat[0].ordered:
```

Comment:
can you do a list comprehension instead,using counters like this is not pythonic